Common/util/path.cpp

```diff
@@ -77,8 +77,8 @@ bool IsSameOrSubDir(const String &parent, const String &path)
     char can_path[MAX_PATH];
     char relative[MAX_PATH];
     // canonicalize_filename treats "." as "./." (file in working dir)
-    const char *use_parent = parent == "." ? "./" : parent;
-    const char *use_path   = path   == "." ? "./" : path;
+    const char *use_parent = (parent == String(".")) ? "./" : parent.GetCStr();
+    const char *use_path   = (path   == String(".")) ? "./" : path.GetCStr();
     canonicalize_filename(can_parent, use_parent, MAX_PATH);
```

Comment:
Since you remove implicit conversion to const char* perhaps this is no longer necessary?
Both AGS::Common::String and std::string implement comparison operator that accepts const char* as an argument, this is to avoid extra string object allocation.

Reply:
Updated this fix.